sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveInspectors.scala

```diff
@@ -666,3 +676,18 @@ private[hive] trait HiveInspectors {
     }
   }
 }
+
+/**
+ * :: DeveloperApi ::
+ * This represents an erased type because of type erasure in JVM.
+ */
+@DeveloperApi
+class ErasedType private() extends DataType {
```

Comment:
I agree with @chenghao-intel that we should confirm this is supported with hive using a compatibility test.  If so, we should probably just use `NullType` here